hudi-sync/hudi-sync-common/src/main/java/org/apache/hudi/sync/common/HoodieSyncTool.java

```diff
@@ -41,6 +43,7 @@ public HoodieSyncTool(Properties props) {
   public HoodieSyncTool(Properties props, Configuration hadoopConf) {
     this.props = props;
     this.hadoopConf = hadoopConf;
+    this.metrics = new HoodieMetaSyncMetrics(new HoodieSyncConfig(props, hadoopConf), getClass().getSimpleName());
```

Comment:
Invoking shutdown on the `Metrics` instance inside `metrics` is required, in case the sync tool is used independently.

Reply:
sync tool is run as part of the StreamSync's runMetaSync method. StreamSync does the shutdown of the Metrics in the close method. Do we still need to shutdown the metrics from here as well?

Reply:
If metrics is shutdown when sync tool is closed. It'll close all the reporters. And when SyncTool is run as part of Hoodie Streamer, once meta sync is complete(metrics are shutdown after closing sync tool) and control comes back to StreamSync, when it pushes any metrics they won't be reported right? since all the reporters are closed.

Let me know if my understanding is incorrect.

Reply:
Had a discussion with Ethan, For now we'll be relying on Metrics shutdown hook to stop the metrics.

Reply:
Synced offline.  There should be a better way to shut down metrics instance from these wrapper metrics classes.  Nevertheless, you're right that there is no need to shut down the metrics instance within `HoodieMetaSyncMetrics` class, and we can rely on `Metrics` shutdown hook to clean up the resources properly.